tool_commands/guess_indicators/test_guess_indicators.py

```diff
@@ -0,0 +1,238 @@
+#     return pipes_id_friction_value
+
+
+def empty_manhole_indicator(session, manholes_pre_empty):
+    """ empty column manhole_indicator (set is to NULL) of table v2_manhole """
+    manholes_ids = list(manholes_pre_empty)
+    up = (
+        update(Manhole)
+        .where(Manhole.id.in_(manholes_ids))
+        .values(manhole_indicator=None)
+    )
+    session.execute(up)
+    session.commit()
+
+
+def empty_pipe_friction_value(session, pipes_pre_empty):
+    """ empty column manhole_indicator (set is to NULL) of table v2_manhole """
+    pipes_ids = list(pipes_pre_empty)
+    up = (
+        update(Pipe)
+        .where(Pipe.id.in_(pipes_ids))
+        .values(friction_value=None, friction_type=None)
+    )
+    session.execute(up)
+    session.commit()
+
+
+@pytest.fixture()
+def tmp_sqlite_path(tmpdir):
+    sqlite_filename = "v2_bergermeer.sqlite"
+    orig_sqlite_path = TEST_DATA_DIR.joinpath(
+        "testmodel", "v2_bergermeer", sqlite_filename
+    )
+    tmp_sqlite_dir = Path(tmpdir)
+    tmp_sqlite_path = tmp_sqlite_dir.joinpath(sqlite_filename)
+    shutil.copy2(orig_sqlite_path, tmp_sqlite_path)
+    return tmp_sqlite_path
+
+
+@pytest.fixture()
+def db(tmp_sqlite_path):
+    db_type = "spatialite"
+    db_set = {"db_path": tmp_sqlite_path}
+    db = ThreediDatabase(db_set, db_type)
+    return db
+
+
+@pytest.fixture()
+def guesser(db):
+    return Guesser(db)
+
+
+def test_guess_manhole_indicator(db, guesser):
+    session = db.get_session()
+
+    # before we empty manholes, first get their [(id, manhole_indicator)]
+    manholes_pumps_pre_empty = get_manholes(session, pump_bool=True)
+    manholes_outlets_pre_empty = get_manholes(session, outlet_bool=True)
+    all_manholes_pre_empty = get_manholes(session, all_bool=True)
+    if not all_manholes_pre_empty:
+        # skip this test as there are no manholes in sqlite..
+        return
+    # now lets empty column 'manhole_indicator' in all manholes
+    empty_manhole_indicator(session, all_manholes_pre_empty)
+
+    all_manholes_pre_guess = get_manholes(session, all_bool=True)
+    # all dict values should be None (<-- still test-prework, not actual testing), If
+    # it fails then empty_manhole_selection() does not a good job
+    assert not all(list(all_manholes_pre_guess.values()))
+
+    # now put guesser to work
+    guesser.guess_manhole_indicator(only_empty_fields=False)
+
+    # get a new session
+    session = db.get_session()
+
+    manholes_pumps_after_guess = get_manholes(session, pump_bool=True)
+    manholes_outlet_after_guess = get_manholes(session, outlet_bool=True)
+    all_manholes_after_guess = get_manholes(session, all_bool=True)
+
+    # manholes_pumps should have been updated to MANHOLE_INDICATOR_PUMPSTATION
+    if manholes_pumps_pre_empty:
+        result_list = list(manholes_pumps_after_guess.values())
+        expected_value = Constants.MANHOLE_INDICATOR_PUMPSTATION
+        assert all([expected_value == ele for ele in result_list])
```

Comment:
Traditioneel word "elem" gebruikt i.p.v. "ele" als er een afkorting nodig is.
Vaker zie ik trouwens "item".

(En liever nog een naam die aangeeft wat het is. Item "result_list". Kan "manholes" en dan "for manhole in manholes" niet gewoon?)